service/src/test/kotlin/io/provenance/api/models/cee/execute/ContractExecutionResponseTest.kt

```diff
@@ -0,0 +1,61 @@
+package io.provenance.api.models.cee.execute
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.shouldBe
+import io.provenance.api.models.p8e.TxResponse
+import io.provenance.scope.encryption.domain.inputstream.DIMEInputStream.Companion.configureProvenance
+import java.util.UUID
+
+class ContractExecutionResponseTest : FunSpec({
+    context("mapping") {
+        val mapper = ObjectMapper().configureProvenance()
+
+        test("SinglePartyContractExecutionResponse") {
+            val response = SinglePartyContractExecutionResponse(
+                type = "single",
+                metadata = TxResponse(
+                    hash = "hash",
+                    gasWanted = "2",
+                    gasUsed = "3",
+                    height = "4"
+                ),
+                error = "error",
+                scopeUuids = listOf(UUID.randomUUID())
+            )
+
+            val serialized = mapper.writeValueAsString(response)
+            val deserialized: ContractExecutionResponse =
+                mapper.readValue(serialized, ContractExecutionResponse::class.java)
+            (deserialized as SinglePartyContractExecutionResponse) shouldBe response
+        }
+
+        test("ContractExecutionErrorResponse") {
+            val response = ContractExecutionErrorResponse(
+                type = "single",
```

Comment:
hmm maybe these `type` properties aren't doing anything because this is the wrong one for this subclass